lib/ActionMetric.js

```diff
@@ -0,0 +1,75 @@
+const { Logger } = require('werelogs');
+
+const MetricsModel = require('./models/MetricsModel');
+
+class ActionMetric {
+    constructor() {
+        this._log = new Logger('ActionMetric');
+    }
+
+    withProducer(producer) {
+        this._producer = producer;
+        return this;
+    }
+
+    withSite(site) {
+        this._site = site;
+        return this;
+    }
+
+    withObjectSize(objectSize) {
+        this._objectSize = objectSize;
+        return this;
+    }
+
+    withEntry(entry) {
```

Comment:
Could maybe add JSDoc explaining that we expect an `ActionQueueEntry` here, since we are calling some of its member functions.